File: typescript/src/renderer.ts
interface Renderer {
    draw: () => void,
    setShape: (newShape: Shape) => void,
}


function newRenderer(shape: Shape): Renderer {
    return {
        draw() {
            const area: number = shape.computeArea()
            console.log("Shape drawn\n" + "Its area is " + area)
        },
        setShape(newShape: Shape) {
            shape = newShape;
        }
    }
}

export { newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer implements IRenderer {
    private Shape shape;
    
    Renderer(Shape shape) {
        this.shape = shape;
    }

    public void draw() {
        double area = shape.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }

    public void setShape(Shape shape) {
        this.shape = shape;
    }
}
